spec.yaml

```
@@ -88,6 +91,115 @@ paths:
+      - name: include_size
+        in: query
+        schema:
+          type: boolean
+
+      - $ref: './components/parameters/Owner.yaml'
+      - $ref: './components/parameters/Repository.yaml'
+      - $ref: './components/parameters/StorageProtocol.yaml'
+      - $ref: './components/parameters/StorageBucket.yaml'
+      - $ref: './components/parameters/Path.yaml'
+      responses:
+        '200':
+          description: Successfully listed a directory
+          content:
+            application/json:
+              schema:
+                $ref: './components/schemas/Files.yaml'
+              example:
+                  [
+                    {
+                      "path": "images/000.png",
+                      "type": "file",
+                      "size": 0,
+                      "hash": "79fb7f8632d7e15b3b46a7411d08bcdd",
+                      "versioning": "bucket",
+                      "download_url": "https://dagshub.com/Simon/baby-yoda-segmentation-dataset/storage/raw/s3/bucket/images/000.png",
+                      "content_url":"https://dagshub.com/Simon/baby-yoda-segmentation-dataset/storage/content/s3/bucket/images/000.png" 
+                    },
+                    {
+                      "path": "images/001.png",
+                      "type": "file",
+                      "size": 0,
+                      "hash": "7dd5befce39375036da253b435e1db98",
+                      "versioning": "bucket",
+                      "download_url": "https://dagshub.com/Simon/baby-yoda-segmentation-dataset/storage/raw/s3/bucket/images/001.png",
+                      "content_url":"https://dagshub.com/Simon/baby-yoda-segmentation-dataset/storage/content/s3/bucket/images/001.png" 
+                    },
+                    {
+                      "path": "images/002.png",
+                      "type": "file",
+                      "size": 0,
+                      "hash": "a58f181cd982a1e21c77949588737c8b",
+                      "versioning": "bucket",
+                      "download_url": "https://dagshub.com/Simon/baby-yoda-segmentation-dataset/storage/raw/s3/bucket/images/002.png",
+                      "content_url":"https://dagshub.com/Simon/baby-yoda-segmentation-dataset/storage/content/s3/bucket/images/002.png" 
+                    },
+                  ]
+        '404':
+          description: Folder or file not found
+
+
+  /repos/{owner}/{repo}/storage/raw/{protocol}/{bucket}/{path}: 
+    get:
+      tags:
+        - Storage Integrations
+      summary: Get file in the bucket
+      operationId: getBucketFile
+      parameters:
+      - $ref: './components/parameters/Owner.yaml'
+      - $ref: './components/parameters/Repository.yaml'
+      - $ref: './components/parameters/StorageProtocol.yaml'
+      - $ref: './components/parameters/StorageBucket.yaml'
+      - name: path
+        in: path
+        required: true
+        description: The content path
+        schema: 
+          type: string
```

```suggestion
      - $ref: './components/parameters/Path.yaml'
```

after you change the description to include buckets

---

1) This one is required, when the /content/ path isn't
2) This is for /raw/, so it shouldn't be folder, it should be a file, meanwhile Path.yaml talks about folders